server/plugin/webhook.go

```diff
@@ -1019,18 +1034,42 @@ func (p *Plugin) handleCommentAssigneeNotification(event *github.IssueCommentEve
 	if len(splitURL) < 2 {
 		return
 	}
+
+	eventType := splitURL[len(splitURL)-2]
 	var templateName string
-	switch splitURL[len(splitURL)-2] {
+	switch eventType {
 	case "pull":
 		templateName = "commentAssigneePullRequestNotification"
 	case "issues":
 		templateName = "commentAssigneeIssueNotification"
 	default:
-		p.API.LogWarn("Unhandled issue type", "type", splitURL[len(splitURL)-2])
+		p.API.LogDebug("Unhandled issue type", "Type", eventType)
 		return
 	}
 
+	mentionedUsernames := parseGitHubUsernamesFromText(event.GetComment().GetBody())
+
 	for _, assignee := range assignees {
+		usernameMentioned := false
+		for _, username := range mentionedUsernames {
+			if username == *assignee.Login {
+				usernameMentioned = true
+				break
+			}
+		}
+
+		if usernameMentioned {
+			switch eventType {
+			case "pull":
+				templateName = "commentAssigneeSelfMentionPullRequestNotification"
+			case "issues":
+				templateName = "commentAssigneeSelfMentionIssueNotification"
+			default:
+				p.API.LogDebug("Unhandled issue type", "Type", eventType)
+				return
+			}
+		}
```

Comment:
We should make sure we set `templateName` back to its previous value for other iterations of this `assignees` loop.

Also this `default` case will never occur I think

Reply:
Removed the default and set the templateName